feat: upgrade ring-jetty to 1.14.1

edpaget opened the pull request:
### Description

Moves us onto Jetty 12 which has some features (pthreading, QoSHanlder) that we may want to take advantage of.

Jetty is not longer a purely servlet-based server so the servlet APIs we had used previously had to be update for the new location of those classes and to use adapters to the core jetty handler implementation. 

### How to verify

This _should_ have no differences to the user. 

### Checklist

- [N/A] Tests have been added/updated to cover changes in this PR

piranha commented on deps.edn:
```diff
@@ -183,10 +180,8 @@
-                                             :exclusions [org.eclipse.jetty/jetty-server
-                                                          org.eclipse.jetty.websocket/websocket-jetty-server]}
+  ring/ring-core                            {:mvn/version "1.14.1"}             ; HTTP abstraction
+  ring/ring-jetty-adapter                   {:mvn/version "1.14.1"}              ; Jetty adapter}
```
comment on this line is slightly weird

sansolovyov reviewed (approved):
Wow, cool! I assumed there would be more to it, but I guess it's not that bad if you know what you're doing. 😁

piranha commented:
Why does it complain about jetty not containing licenses :'(

edpaget committed:
feat: upgrade ring-jetty to 1.14.1

Moves us onto Jetty 12 which has some features (pthreading, QoSHanlder)
that we may want to take advantage of.

edpaget commented:
There's a still something with prometheus to untangle, and I had to bump Jetty to the alpha version of the unrelease 12.1.0 because there is some error which spams the logs in the [EE9 servlet handler](https://github.com/jetty/jetty.project/issues/12603). It looks like it'd be pretty straightforward to port our streaming handlers to Jetty's core handler system and skip Servlet entirely.

edpaget committed:
fixup! customize jetty metrics

edpaget committed:
fixup! license info for jetty ee9

edpaget committed:
fixup! make a var private

edpaget committed:
fixup! don't use deprecated method

edpaget committed:
fixup! create custom statistics handler

Replicate the Jetty 11 statistics handler that is able to report on
asynchronously handled requests separately from synchronously handled
ones. Also ports it to directly report statistics to our prometheus
rather than needing to build a connector between the stats handler
object and our metrics reporting

Co-Authored-By: John Swanson <john.swanson@metabase.com>

edpaget committed:
Merge branch 'master' into ep-upgrade-to-ring-jetty-1-14

edpaget committed:
fixup! fix race condition in stats test

edpaget committed:
fixup! missing private annotation

edpaget committed:
fixup! use normal jetty release with logging fix

paoliniluis commented:
@edpaget saw you added support for unix sockets and websockets, is that correct?

paoliniluis commented:
I'm asking since I see org.eclipse.jetty/jetty-unixdomain-server and org.eclipse.jetty.ee9.websocket/jetty-ee9-websocket-jetty-server in the deps, which we didn't have before

piranha commented:
@paoliniluis that's just the different way they split their package, both unix domains and websockets were supported before.

@edpaget 🎉🎉🎉
